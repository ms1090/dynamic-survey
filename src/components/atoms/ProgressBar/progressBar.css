.progress-bar {
  background:#fff;
  border:1px solid #97c759;
  border-radius: 3px;
  color:#97c759;
  height: 10px;
  margin: 10px 0 20px;
  width:100%;
}
.progress-bar::-webkit-progress-bar { background:#fff; }
.progress-bar::-webkit-progress-value,
progress::-moz-progress-bar { background:#97c759; }
