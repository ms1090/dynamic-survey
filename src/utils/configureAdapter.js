import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
