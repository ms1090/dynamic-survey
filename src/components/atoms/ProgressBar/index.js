import React from 'react';
import PropTypes from 'prop-types';
import './progressBar.css';

/**
 * Display a progress element as per customization props
 * @param {String} className CSS class name
 * @return {React.Element} Return progress element UI
 */
const ProgressBar = ({
  className,
  ...others
}) => (
  <progress
    className={`progress-bar ${className}`}
    {...others}
  />
);
// default values for component props
ProgressBar.defaultProps = {
  className: '',
};
// prop types checking
ProgressBar.propTypes = {
  className: PropTypes.string,
};
export default ProgressBar;
