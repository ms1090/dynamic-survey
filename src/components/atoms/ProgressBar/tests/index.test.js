import React from 'react';
import { shallow } from 'enzyme';
import ProgressBar from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('ProgressBar Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<ProgressBar className="test-className" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
