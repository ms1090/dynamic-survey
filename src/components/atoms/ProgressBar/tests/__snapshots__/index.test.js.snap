// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component should render correctly 1`] = `
<progress
  className="progress-bar test-className"
/>
`;
